File: tests/fixtures/dotnet/dirty.cs
// Deliberately unsafe/broken fixture — exercises every dotnet.js finding id.
//
// Both files here declare `namespace Fixtures;` (the file-scoped form) — a
// real C# project would collide if two files in one directory declared the
// same namespace *and* the same class name, so DirtyExamples and
// CleanExamples (in clean.cs) are kept distinct even though the namespace is
// shared. File-scoped namespace syntax also avoids an extra brace level of
// pure structural nesting that would otherwise trip the shared
// obvious/nesting-depth check on scoping alone, before any real logic.
namespace Fixtures;

using System;
using System.Data.SqlClient;
using System.Runtime.Serialization.Formatters.Binary;
using System.Security.Cryptography;

class DirtyExamples
{
    void LookupUser(SqlConnection conn, string id)
    {
        var cmd = new SqlCommand($"SELECT * FROM t WHERE id = {id}", conn);
        cmd.CommandText = "SELECT * FROM t WHERE id = " + id;
    }

    void Deserialize(System.IO.Stream payload)
    {
        var f = new BinaryFormatter();
        f.Deserialize(payload);
    }

    void HashPassword(string password)
    {
        MD5.Create();
    }

    string MakeToken()
    {
        var token = new Random().Next().ToString();
        return token;
    }

    void InsecureHandler()
    {
        ServicePointManager.ServerCertificateValidationCallback = (a, b, c, d) => true;
    }

    void Swallow()
    {
        try
        {
            Go();
        }
        catch (Exception)
        {
        }
    }

    void DebugPrint(int x)
    {
        Console.WriteLine("here " + x);
    }

    void RunGitLog(string branch)
    {
        Process.Start($"git log {branch}");
    }

    void RunShell(string cmd)
    {
        var psi = new ProcessStartInfo { FileName = "cmd.exe", Arguments = $"/c {cmd}", UseShellExecute = true };
        Process.Start(psi);
    }

    void Go()
    {
    }
}

File: tests/fixtures/py/dirty.py
# Deliberately unsafe/broken fixture — exercises every py.js finding id.

import hashlib
import os
import pickle
import subprocess
import yaml


def lookup_user(cursor, uid):
    return cursor.execute(f"SELECT * FROM t WHERE id = {uid}")


def run_command(cmd, target):
    os.system("rm " + target)
    subprocess.run(cmd, shell=True)


def run_payload(user_input):
    eval(user_input)


def load_blob(payload):
    return pickle.loads(payload)


def hash_password(password):
    return hashlib.md5(password.encode())


def insecure_request(session):
    session.get("https://example.com", verify=False)


def add(item, into=[]):
    into.append(item)
    return into


def bare():
    try:
        go()
    except:
        pass


def swallowed():
    try:
        go()
    except Exception:
        pass


def debug_print():
    print("here")
    breakpoint()


def deep(x, y, z, w):
    if x:
        for a in y:
            while z:
                go(w)


def load_profile(url):
    s = requests.Session()
    return s.get(url)


def lookup_user_tainted(cursor, uid):
    q = f"SELECT * FROM t WHERE id = {uid}"
    return cursor.execute(q)
